e2e_test/iceberg/test_case/iceberg_engine.slt

```diff
@@ -113,3 +113,46 @@ select count(_row_id) from t_without_pk;
 
 statement ok
 DROP TABLE t_without_pk
+
+# test iceberg connection in iceberg engine
+statement ok
+create secret my_secret with (
+  backend = 'meta'
+) as 'hummockadmin';
+
+statement ok
+create connection my_conn
+with (
+    type = 'iceberg',
+    warehouse.path = 's3://hummock001/iceberg_connection',
+    s3.access.key = secret my_secret,
+    s3.secret.key = secret my_secret,
+    s3.endpoint = 'http://127.0.0.1:9301',
+    s3.region = 'us-west-2',
+);
+
+statement ok
+set iceberg_engine_connection = 'public.my_conn';
```

Comment:
Why don't we use the `connection = conn` syntax, but use a session variable instead? :thinking

Reply:
Finding a way to allow users to set a default behavior for a whole database. Adding a syntax is also ok, however, our iceberg table can used with a connector which also can have a connection. Maybe a little bit confused to the user.

Reply:
Oh, that's indeed a problem... So the `with` can be both for connector and for the table.

Reply:
crunchybridge https://arc.net/l/quote/vdcxffua

Reply:
Oh, we will allow user to specify both volume and catalog (or any 1 of them) in the same connection, right?

Reply:
yes. iceberg connection can specify catalog and bucket together. Although for iceberg engine, we only allow specifying bucket